Source/JavaScriptCore/wasm/WasmFunctionParser.h

```diff
@@ -1629,6 +1629,54 @@ auto FunctionParser<Context>::parseArrayTypeDefinition(const char* operation, bo
     return { };
 }
 
+// Return true if parsing should continue after, false if parseExpression() should return.
+template <typename Context>
+ALWAYS_INLINE bool FunctionParser<Context>::parseNestedBlocksEagerly()
+{
+    do {
+        int8_t kindByte;
+        BlockSignature inlineSignature;
+
+        // Only attempt to parse the most optimistic case of a single non-ref or void return signature.
+        if (LIKELY(peekInt7(kindByte) && isValidTypeKind(kindByte))) {
+            TypeKind typeKind = static_cast<TypeKind>(kindByte);
+            Type type = { typeKind, TypeDefinition::invalidIndex };
+            if (UNLIKELY(!(type.isVoid() || isValueType(type))))
+                return true;
+            inlineSignature = m_typeInformation.thunkFor(type);
+        } else
+            return true;
+
+        ASSERT(!inlineSignature->argumentCount());
+
+        int64_t oldSize = m_expressionStack.size();
+        Stack newStack;
+        ControlType block;
+        auto result = m_context.addBlock(inlineSignature, m_expressionStack, block, newStack);
```

Comment:
Can this actually fail, but later succeed below? I think we should use WASM_PARSER_FAIL_IF so it is clear that this is doing the same thing as the code below, and we should decide to continue only if this function fails. Idk, what do you think?

Reply:
It should either fail together or succeed together, I just wrote it like this originally to avoid doing a more complex error check in the fast path. I think it's probably ok to just return a `PartialResult` from this helper though, so I've changed the continue flag to be a bool reference that's passed in.

(the other fail cases are still relegated to the slow path, because it involves constructing an error)